test-result-summary-client/src/Build/AllTestsInfo.jsx

```diff
@@ -64,27 +90,37 @@ export default class Build extends Component {
             } );
         }
 
-        testData.sort(( a, b ) => {
+        testResult.sort(( a, b ) => {
             let rt = a[0].testResult.localeCompare( b[0].testResult );
             if ( rt === 0 ) {
                 return a.key.localeCompare( b.key );
             }
             return rt;
         } );
 
+        const { testData } = this.state;
         this.setState( {
             parents: builds.map( element => { return { buildNum: element.parentNum, timestamp: element.parentTimestamp }; } ),
-            testData,
+            testData: testData.concat(testResult),
         } );
     }
 
     render() {
         const { testData, parents } = this.state;
-        const { buildId } = getParams( this.props.location.search );
-        return <div>
-            <TestBreadcrumb buildId={buildId} />
-            <SearchOutput buildId={buildId} />
-            <TestTable title={"Tests"} testData={testData} parents={parents} buildId={buildId} />
-        </div>
+        const { buildId, parentId } = getParams( this.props.location.search );
+        if (parentId) {
+            return <div>
+                <TestBreadcrumb buildId={parentId} />
+                <SearchOutput buildId={parentId} />
+                <TestTable title={"Tests"} testData={testData} parents={parents} />
+            </div>
+
+        } else {
+            return <div>
+                <TestBreadcrumb buildId={buildId}/>
+                <SearchOutput buildId={buildId}/>
+                <TestTable title={"Tests"} testData={testData} parents={parents} />
+            </div>
+        }
```

Comment:
I think this code can be reverted back as we pass in `buildId ` only (see comment in ResultGrid.jsx)